### PR TITLE
Http api merged from th0br0

### PR DESCRIPTION
**Pull request by tsvisabo**
- Fork and resolve conflicts

### CODE REVIEW

File: cclient/serialization/json/get_node_info.c [hunk 1]
```diff
@@ -9,99 +9,157 @@
 #include "cclient/serialization/json/helpers.h"
 #include "cclient/serialization/json/logger.h"
 
+static const char *kAppName = "appName";
```
Comment (thibault-martinez) [inline cclient/serialization/json/get_node_info.c]: const placement for this and all following

Comment (thibault-martinez) [reply cclient/serialization/json/get_node_info.c]: Also, snake_case

File: ciri/api/api.h [hunk 2]
```diff
@@ -34,12 +36,8 @@ extern "C" {
  */
 typedef struct iota_api_s {
   iota_api_conf_t conf;
-  thread_handle_t thread;
   bool running;
```
Comment (thibault-martinez) [inline ciri/api/api.h]: This is now useless

File: cclient/serialization/json/get_node_info.c (hunk 1 above)
Comment (oopsmonk) [inline cclient/serialization/json/get_node_info.c]: should be `static char const *`

File: cclient/serialization/json/get_node_info.c [hunk 3]
```diff
@@ -9,99 +9,157 @@
...
+  size_t len = 0;
+  log_debug(json_logger_id, "[%s:%d]\n", __func__, __LINE__);
+
+  cJSON *json_root = cJSON_CreateObject();
+  if (json_root == NULL) {
+    log_critical(json_logger_id, "[%s:%d] %s\n", __func__, __LINE__, STR_CCLIENT_JSON_CREATE);
+    return RC_CCLIENT_JSON_CREATE;
+  }
+
+  cJSON_AddStringToObject(json_root, kAppName, obj->app_name->data);
```
Comment (oopsmonk) [inline cclient/serialization/json/get_node_info.c]: now we have provied access metnods in response/request, ex.`get_node_info_res_app_name()`

File: ciri/api/http.c [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+    return RC_NULL_PARAM;
+  } else if (api->running == false) {
+    return RC_OK;
+  }
+
+  MHD_stop_daemon(api->state);
+
+  for (tangle = (tangle_t *)utarray_back(api->db_connections); tangle != NULL;
+       tangle = (tangle_t *)utarray_back(api->db_connections)) {
+    log_error(logger_id, "Destroying HTTP API database connection\n");
```
Comment (thibault-martinez) [inline ciri/api/http.c:229]: log_debug

File: cclient/serialization/json/get_node_info.c [hunk 5]
```diff
@@ -9,99 +9,157 @@
...
+
+  ret = flex_trits_to_json_string(json_root, kCoordinatorAddress, obj->coordinator_address, HASH_LENGTH_TRIT);
+  if (ret != RC_OK) {
+    goto done;
+  }
+
+  json_text = cJSON_PrintUnformatted(json_root);
+  if (json_text) {
+    len = strlen(json_text);
+    ret = char_buffer_allocate(out, len);
```
Comment (oopsmonk) [inline cclient/serialization/json/get_node_info.c]: `char_buffer_set` is better in this case.

File: cclient/serialization/json/tests/get_node_info.c [hunk 6]
```diff
@@ -68,12 +69,16 @@ void test_deserialize_get_node_info(void) {
...
+
   get_node_info_res_free(&node_info);
+  char_buffer_free(serializer_out);
 }
 int main(void) {
   UNITY_BEGIN();
 
-  RUN_TEST(test_serialize_get_node_info);
-  RUN_TEST(test_deserialize_get_node_info);
+  RUN_TEST(test_request);
```
Comment (oopsmonk) [inline cclient/serialization/json/tests/get_node_info.c]: Could we use `test_serialize_get_node_info` and `test_deserialize_get_node_info` for the consistncy?

**Review by oopsmonk — CHANGES_REQUESTED** [review]

**Review by thibault-martinez — DISMISSED** [review]

**Review by oopsmonk — APPROVED** [review]

**Review by oopsmonk — CHANGES_REQUESTED** [review]
need to fix it.

**Review by oopsmonk — DISMISSED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**